sys/net/application_layer/coap/gcoap.c

```diff
@@ -135,7 +135,7 @@ static void _listen(sock_udp_t *sock)
         return;
     }
 
-    if (coap_get_code(&pdu) == COAP_CODE_EMPTY) {
+    if (pdu.hdr->code == COAP_CODE_EMPTY) {
```

Comment:
Yes, as inline we could add both and use what is appropriate.  So, ACK on this for now with the intention to add to nanocoap. I just added Issue [#19](https://github.com/kaspar030/sock/issues/19) to sock.